html/pfappserver/lib/pfappserver/Form/Config/Provisioning/mobileconfig.pm

```diff
@@ -8,27 +8,90 @@ pfappserver::Form::Config::Provisioning - Web form for a switch
 
 =cut
 
+use pf::config;
 use HTML::FormHandler::Moose;
 extends 'pfappserver::Form::Config::Provisioning';
 with 'pfappserver::Base::Form::Role::Help';
 
+has_field 'company' =>
+  (
+   type => 'hidden',
+   value => $Config{'pki'}{'organisation'},
+  )
+
+has_field 'fingerprint' =>
+  (
+   type => 'hidden'.
+   value => $data,
+  )
+
 has_field 'ssid' =>
   (
    type => 'Text',
    label => 'SSID',
   );
 
-has_field 'ca_cert_path' =>
+has_field 'security_type' =>
+  (
+   type => 'Select',
+   multiple => 0,
+   label => 'Security type',
+   options_method => \&option_security,
+   element_class => ['chzn-deselect'],
+   tags => { after_element => \&help,
+             help => 'Select the type of security applied for your SSID' },
+  );
+
+has_field 'eap_type' =>
+  (
+   type => 'Select',
+   multiple => 0,
+   label => 'EAP type',
+   options_method => \&options_eap_type,
+   element_class => ['chzn-deselect'],
+   tags => { after_element => \&help,
+             help => 'Select the EAP type of your SSID' },
+  );
+
+has_field 'passcode' =>
+  (
+   type => 'Text',
+   label => 'Wifi Key',
+  );
+
+has_field 'reversedns' =>
   (
-   type  => 'Text',
-   label => 'Certificate',
+   type => 'Text',
+   label => 'ReverseDNS identifier',
+   tags => { after_element => \&help,
+             help => 'Example : org.packetfence.www'},
```

Comment:
Good that you write an example but you're not doing the normal to reverse, you're just writing the reverse.

This assumes I know what PacketFence's website is so I know how to build the reverse